packages/dds/tree/src/simple-tree/api/verboseTree.ts

```diff
@@ -0,0 +1,409 @@
+import type {
+	TreeLeafValue,
+	ImplicitFieldSchema,
+	ImplicitAllowedTypes,
+} from "../schemaTypes.js";
+import { getSimpleNodeSchema, NodeKind, type TreeNodeSchema } from "../core/index.js";
+import {
+	isTreeValue,
+	stackTreeFieldCursor,
+	stackTreeNodeCursor,
+	type CursorAdapter,
+} from "../../feature-libraries/index.js";
+import {
+	booleanSchema,
+	handleSchema,
+	nullSchema,
+	numberSchema,
+	stringSchema,
+} from "../leafNodeSchema.js";
+import { toFlexSchema } from "../toFlexSchema.js";
+import { isObjectNodeSchema } from "../objectNodeTypes.js";
+import { walkFieldSchema } from "./tree.js";
+
+/**
+ * Verbose encoding of a {@link TreeNode} or {@link TreeValue}.
+ * @remarks
+ * This is verbose meaning that every {@link TreeNode} is a {@link VerboseTreeNode}.
+ * Any IFluidHandle values have been replaced by `THandle`.
+ * @privateRemarks
+ * This can store all possible simple trees, but it can not store all possible trees representable by our internal representations like FlexTree and JsonableTree.
+ */
+export type VerboseTree<THandle = IFluidHandle> =
+	| VerboseTreeNode<THandle>
+	| Exclude<TreeLeafValue, IFluidHandle>
+	| THandle;
+
+/**
+ * Verbose encoding of a {@link TreeNode}.
+ * @remarks
+ * This is verbose meaning that every {@link TreeNode} has an explicit `type` property, and `fields`.
+ * This allowed VerboseTreeNode to be unambiguous regarding which type each node is without relying on symbols or hidden state.
+ *
+ * Any IFluidHandle values have been replaced by `THandle`. If the `THandle` is JSON compatible, then this type is JSON compatible as well.
+ *
+ * @privateRemarks
+ * This type is only used for data which is copied into and out of the tree.
+ * When being copied out, its fine to have the data be mutable since its a copy.
+ *
+ * When being copied in, we don't need to mutate, so we could use a readonly variant of this type.
+ * however the copy in case (createFromVerbose) probably isn't harmed much by just reusing this type as is,
+ * since if the caller has immutable data, TypeScript doesn't prevent assigning immutable data to a mutable type anyway.
+ * Also relaxing the input methods to take readonly data would be a non-breaking change so it can be done later if desired.
+ *
+ * This format is simple-tree specialized alternative to {@link JsonableTree}.
+ * This format allows for all simple-tree compatible trees to be represented.
+ *
+ * Unlike `JsonableTree`, leaf nodes are not boxed into node objects, and instead have their schema inferred from the value.
+ * Additionally, sequence fields can only occur on a node that has a single sequence field (with the empty key) replicating the behavior of simple-tree ArrayNodes.
+ */
+export interface VerboseTreeNode<THandle = IFluidHandle> {
+	/**
+	 * The meaning of this node.
+	 * Provides contexts/semantics for this node and its content.
+	 * Typically used to associate a node with metadata (including a schema) and source code (types, behaviors, etc).
```

Comment:
Nit: this last line probably belongs in a `@remarks`, since it describes potential usage. An example would also probably be useful.